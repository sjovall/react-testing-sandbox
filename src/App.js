import React, {Component} from 'react';

class App extends Component {

    constructor(props) {
        super(props);
        this.state = {counter: 0};
    }

    onIncrement = () => this.setState(doIncrement);
    onDecrement = () => this.setState(doDecrement);

    render() {
        const {counter} = this.state;

        return (
            <div>
                <h1>My Counter</h1>
                <Counter counter={counter}/>

                <button type="button" onClick={this.onIncrement}>
                    Increment
                </button>

                <button type="button" onClick={this.onDecrement}>
                    Decrement
                </button>
            </div>
        );
    }
}


export const Counter = ({counter}) => (
    <p>{counter}</p>
);

const doIncrement = (prevState) => ({
    counter: prevState.counter + 1,
});

const doDecrement = (prevState) => ({
    counter: prevState.counter - 1,
});

export default App;