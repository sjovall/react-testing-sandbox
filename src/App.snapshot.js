import React from 'react';
import App, { Counter } from './App';
import renderer from 'react-test-renderer';


describe('App snapshot', () => {
   test('renders', () => {
       const component = renderer.create(<App/>);

       let tree = component.toJSON();
       expect(tree).toMatchSnapshot();
   });

   test('renders', () => {
       const component = renderer.create(<Counter counter={2}/>);

       let tree = component.toJSON();
       expect(tree).toMatchSnapshot();
   })
});