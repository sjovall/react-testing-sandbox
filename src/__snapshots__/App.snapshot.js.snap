// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div>
  <h1>
    My Counter
  </h1>
  <h2>
    Waffles
  </h2>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Increment
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Decrement
  </button>
</div>
`;

exports[`App snapshot renders 2`] = `
<p>
  2
</p>
`;
